spec/unit/installer/user_project_integrator/target_integrator_spec.rb

```diff
@@ -280,6 +283,53 @@ module Pod
           phase = target.shell_script_build_phases.find { |bp| bp.name == @embed_framework_phase_name }
           phase.nil?.should == true
         end
+
+        it 'does not add embed frameworks build phase input output paths with no frameworks' do
+          @pod_bundle.stubs(:frameworks_by_config => {})
+          @target_integrator.integrate!
+          target = @target_integrator.send(:native_targets).first
+          phase = target.shell_script_build_phases.find { |bp| bp.name == @embed_framework_phase_name }
+          phase.input_paths.should.be.empty
+          phase.output_paths.should.be.empty
+        end
+
+        it 'adds embed frameworks build phase input and output paths for vendored and non vendored frameworks' do
+          debug_vendored_framework = { :framework => '${PODS_ROOT}/DebugVendoredFramework/ios/DebugVendoredFramework.framework',
```

Comment:
vendored dynamic frameworks are copied straight from `PODS_ROOT` versus the compiled ones are placed by Xcode into `BUILT_PRODUCTS_DIR`.

This test covers both cases for vendored artifacts and compiled artifacts.